crates/aptos/src/move_tool/fmt.rs

```diff
@@ -108,71 +113,114 @@ impl FmtCommand {
             cmd
         };
         let to_cli_error = |e| CliError::IO(exe.display().to_string(), e);
-        let package_opt = if let Some(path) = package_opt {
-            fs::canonicalize(path.as_path()).ok()
-        } else {
-            None
-        };
-        let package_path = dir_default_to_current(package_opt.clone())?;
-        let root_res = SourcePackageLayout::try_find_root(&package_path.clone());
-        if let Ok(root_package_path) = root_res {
-            let mut path_vec = vec![];
-            let sources_path = root_package_path.join(SourcePackageLayout::Sources.path());
-            if sources_path.exists() {
-                path_vec.push(sources_path.clone());
-            }
-            let scripts_path = root_package_path.join(SourcePackageLayout::Scripts.path());
-            if scripts_path.exists() {
-                path_vec.push(scripts_path.clone());
-            }
-            let tests_path = root_package_path.join(SourcePackageLayout::Tests.path());
-            if tests_path.exists() {
-                path_vec.push(tests_path.clone());
+
+        // Get the list of files to format
+        let mut files_to_format = Vec::new();
+
+        // Add individual files if provided
+        if let Some(files) = files_opt {
+            // Verify all files exist
+            for file in &files {
+                if !file.exists() {
+                    return Err(CliError::UnexpectedError(format!(
+                        "File does not exist:{:?}",
+                        file
+                    )));
+                }
+                if !file.is_file() {
+                    return Err(CliError::UnexpectedError(format!(
+                        "Path is not a file:{:?}",
+                        file
+                    )));
+                }
+                if file.extension().unwrap() != move_command_line_common::files::MOVE_EXTENSION {
+                    return Err(CliError::UnexpectedError(format!(
+                        "File is not a Move file:{:?}",
+                        file
+                    )));
+                }
             }
-            let examples_path = root_package_path.join(SourcePackageLayout::Examples.path());
-            if examples_path.exists() {
-                path_vec.push(examples_path.clone());
+            files_to_format.extend(files);
+        }
+
+        // Add package files if provided. If no individual files were specified and no package path was provided,
+        // this treats the current directory as the package.
+        if package_opt.is_some() || files_to_format.is_empty() {
```

Comment:
IMHO, it might make more sense to make option of package and files mutually exclusive.

Reply:
Sure. I've been going back and forth between this. I actually initially implemented it this way. I see value in both; no hard preference.